src/client.ts

```diff
@@ -294,13 +296,16 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
           success: false,
           reason: 'TIMEOUT',
           message:
-            'failed to initialize onReady before timeout, either the datafile or user info was not set before the timeout',
+            'failed to initialize onReady before timeout, either the datafile, user info or qualified segments was not set before the timeout',
           dataReadyPromise: this.dataReadyPromise,
         });
       }, timeout) as any;
     });
 
-    return Promise.race([this.dataReadyPromise, timeoutPromise]).then(res => {
+    return Promise.race([this.dataReadyPromise, timeoutPromise]).then(async res => {
+      if (res.success) {
+        await this.fetchQualifiedSegments(config?.segmentOptions);
```

Comment:
What if fetch fails with network error, timeout, etc? res should be updated accordingly?